### ISSUE TITLE
Delay working uncorrectly

### ISSUE THREAD

**Issue by MetalGearDaner** [FlyingPikachu/Quests#67]
Whenever users finish a Quests, they quit from the server and join again, they try to activate the Quest again (this quest has 24h of delay) and the CAN, so that they avoid the delay.

Staff player have tested this and it seems that this problem doesn't appear if you are admin or mod. There are not any permission from Quests set on any rank. Can anybody give me a solution?

**Comment by FlyingPikachu** [comment]
Cannot duplicate. Post your config.yml and quests.yml?

**Comment by MetalGearDaner** [comment]
Here is the Quests.yml:
http://paste.ofcode.org/RhFT24ChHu8LcYLut9B6uG

Here is the config.yml:
http://paste.ofcode.org/WZEQWrKrEeXtF6QNQegYA9

**Comment by FlyingPikachu** [comment]
Still cannot duplicate. I was able to take the quest Fuel10 and relog without the delay resetting.

http://imgur.com/3JSIRJW

What are the "max-party-size" and "party-invite-timeout" nodes in config.yml from? Are they from another plugin? Or perhaps you running an older version of Quests?

**Comment by Formula350** [comment]
Just before 2.0, Blackvein had been ramping up to allow Quest Parties/Groups (something I was looking forward to), but a bunch of the config options went missing in 2.0, and so I assumed that either it wasn't working right or he decided to scrap it.... I suspect MetalGearDaner's config may be quite old and they are simply holdovers from that time.

**Comment by MetalGearDaner** [comment]
Maybe regenerating the config file, this problem should be solve. I'll try within some hours

**Comment by MetalGearDaner** [comment]
I've regenerated the config and problem persist. I've been cheking everything and it seems to be a problem located on the folder Quests/data. That folder contains all the player's UUIDs. Here is an example of a player who experiments the problem:

http://paste.ofcode.org/U66zaefRhirJHNepmxQL5q

**Comment by MetalGearDaner** [comment]
Hello? I need help with this, I have my server in stand by mode because of this problem : s

**Comment by FlyingPikachu** [comment]
I see that player's 'completed-quests' node has several 'null' values. Try removing those.

**Comment by MetalGearDaner** [comment]
I've removed them, and whenever the player finishes the Quest again, instead of adding to 'completedRedoableQuests' that Quest (wich is redoable), it adds to 'completed-quests' a "null" value.

**Comment by FlyingPikachu** [comment]
Which quest are you testing with (please repost your quests.yml, it 404'd)? I was able to complete the default redo-able quest, Stone Miner, and have it added to completedRedoableQuests as usual.

**Comment by MetalGearDaner** [comment]
Here is the entire current Quests folder: https://mega.co.nz/#!UpgBmK7K!NMyDTorozrbh1RAfUSeipRv6KB3WtRx62H1qyLrQX-Y

We are testing with Jobs01.

**Comment by FlyingPikachu** [comment]
Jobs01 does not have any stages, as indicated by

    stages:
      ordered:
        '1': {}

**Comment by MetalGearDaner** [comment]
Woops, Jobs02. Try testing with the Quest Jobs02.

**Comment by MetalGearDaner** [comment]
Woops, try testing Jobs02 but removing the value of item required for the Quest. As a temporal solution, we have set the requirement of having a custom item wich can only be given by Staff members in order to start any delay-Quest, so that you cannot exploit the problem.

Remove the item requirement and try to test the quest with all the other files located on the folder i've uploaded to MEGA. You should have the same problem as us if it has something to do with the files of the plugin.

**Comment by MetalGearDaner** [comment]
Hello?

**Comment by FlyingPikachu** [comment]
Sorry for the late response. I tried your quest with and without the item requirement and was still blocked by the delay. I'm unfortunately out of ideas.

http://i.imgur.com/kX4FPYF.png

**Comment by MetalGearDaner** [comment]
Well... I've seen that the plugins adds a null value instead of the quest name, any idea about why?

**Comment by FlyingPikachu** [comment]
Afraid not - it didn't add a null value for me. I'll continue investigating. Perhaps you could try deleting and re-making the quest?

**Comment by MetalGearDaner** [comment]
Removing this, and restarting the testing process seems to solve the problem. I keep investigating, but I hardly think that it has something to do with the way the plugin saves the information of a completed redoable quest and its corresponding delay time.

 completedRedoableQuests:
...
completedQuestTimes:
...

(... are the list of quests)

**Comment by Neoblade298** [comment]
So this is a screenshot of a regular player's quest data during the glitch process:

http://gyazo.com/f9421e5e878128dd72695e1248229015
http://gyazo.com/a6d4fc6e059b7747b1f63277ef1a12a1
http://gyazo.com/d299268dc2945855454c69b6c96bb50d

The first screenshot shows the quest data before the player has taken the quest. At this point in time, the quest is off cooldown. The quest in particular he is doing is called 'Killing Ratface'. Note there are 4 different numbers in CompletedQuestTimes.

In the second screenshot, the player has taken the quest 'Killing Ratface'. Nothing has changed in the file at this point except for the top where it shows the player having the quest in progress.

The third screenshot is most interesting. First of all, on Completed Quests, 'null' is added to the Completed Quests. Second of all, one of the 4 CompletedQuestTimes changes its number INSTEAD OF adding on a 5th number (I'm not really sure how that section works, I'm assuming it adds a number for each quest's coodlown). 

I'm not sure if this information gives you anything, but if you reply to this I'll be glad to go set up a quests.yml that you can put on a test server really quickly to duplicate the results.

**Comment by Neoblade298** [comment]
Another thing I just found. When the player leaves and rejoins, this shows up in console:
http://gyazo.com/ca5a7b6b3e329d5cb93e9ed98d5fe9be
That's where "null" appears.

**Comment by FlyingPikachu** [comment]
The null values do seem to be the indicator of the issue (and potentially several other tickets). To clarify, the 3rd screenshot is from completing the quest? Would you be able to post your quests.yml for testing?

**Comment by Neoblade298** [comment]
http://pastebin.com/dRgCSCfs

This is the entire quests.yml with absolutely nothing changed. I didn't remove the excess quests for fear that something would change if I did.
The important things you should note:
- Killing Sewer Spiders, Killing Sewer Skeletons, Killing Sewer Zombies, Killing Cultists, Killing Angvoth, Killing Spider Queen, Killing Ratface are the only repeatable quests out of all these.
- Out of all of these, only Killing Ratface and Killing Cultists have been glitchable
- Killing Sewer Spiders, Killing Sewer Skeletons, Killing Sewer Zombies have been tested and are not glitchable
- I have not tested Killing Angvoth and Killing Spider Queen for glitchability yet but I will if you need me to.

So I guess it has something to do with whatever difference there is between the glitchable and unglitchable quests.

**Comment by MetalGearDaner** [comment]
Any information?

**Comment by FlyingPikachu** [comment]
Bug #57 may be have been related to this issue and has since been resolved. Please update to at least this dev build and check if it resolves the issue.

http://ci.ac3-servers.eu/job/Quests/81/

**Comment by MetalGearDaner** [comment]
This bug isn't fixed with version 2.5.1-b82. Hope can be fixed soon, 4 months since I reported this bug, 4 months since my server can't be played correctly.

**Comment by FlyingPikachu** [comment]
@MetalGearDaner Unfortunately development is slow and I have no clues as to what is causing the 'null' values. The 2 quests that @Neoblade298 was able to glitch don't have any noticeable differences from the ones that weren't glitch-able.

**Comment by FlyingPikachu** [comment]
Note to future debuggers: perhaps #215 is the cause?

**Comment by FlyingPikachu** [comment]
I am choosing to close this issue. Once again I just attempted to replicate the issue (with Quests 2.6.2) to no avail. As a significant amount of time has passed, it is possible that the bug may behave differently now or no longer exist. If you are reading this and experiencing this bug, please create a new ticket including your _entire_ Quests folder. Thanks!
